lib/rust/data-structures/src/hash_map_tree.rs

```diff
@@ -355,6 +355,67 @@ where K: Eq + Hash
     }
 }
 
+impl<K, V, S> HashMapTree<K, Option<V>, S>
+where
+    K: Eq + Hash,
+    S: BuildHasher + Default,
+{
+    /// Sets the value at `path` to `value` and returns the replaced value (returns `None` if there
+    /// was no node at `path`). Then visits nodes on the `path` in reverse order and removes every
+    /// visited empty leaf node from its parent.
+    ///
+    /// A node is defined as empty when it contains a `None` value. A node is a leaf when its
+    /// [`is_leaf`] method returns `true`.
+    ///
+    /// The function is optimized to not create new empty nodes if they would be deleted by the
+    /// function before it returns.
+    pub fn replace_value_and_traverse_back_pruning_empty_leaf<P, I>(
+        &mut self,
+        path: P,
+        value: Option<V>,
+    ) -> Option<V>
+    where
+        P: IntoIterator<Item = I>,
+        I: Into<K>,
+    {
+        let mut path_iter = path.into_iter();
+        let mut swapped_value = value;
+        self.swap_value_and_traverse_back_pruning_empty_leaf(&mut path_iter, &mut swapped_value);
+        swapped_value
+    }
```

Comment:
`A node is defined as empty when it contains a `None` value.` This is incorrect. This is like a `HashMap<K, V>`. Users of this library can keep `V` as `Option<T>` and you can't then consider the leaves of this tree to be empty if the value is `None`. This mixes the internal logic and the API for the user. This can't work this way and this function should be removed from the API.